parquet/src/geospatial/bounding_box.rs

```diff
@@ -0,0 +1,429 @@
+    pub fn with_yrange(mut self, ymin: f64, ymax: f64) -> Self {
+        self.ymin = ymin;
+        self.ymax = ymax;
+        self
+    }
+
+    /// Creates a new bounding box with the specified Z-coordinate range.
+    pub fn with_zrange(mut self, zmin: f64, zmax: f64) -> Self {
+        self.zmin = Some(zmin);
+        self.zmax = Some(zmax);
+        self
+    }
+
+    /// Creates a new bounding box with the specified M-coordinate range.
+    pub fn with_mrange(mut self, mmin: f64, mmax: f64) -> Self {
+        self.mmin = Some(mmin);
+        self.mmax = Some(mmax);
+        self
+    }
+
+    /// Returns the minimum x-coordinate.
+    pub fn get_xmin(&self) -> f64 {
+        self.xmin
+    }
+
+    /// Returns the maximum x-coordinate.
+    pub fn get_xmax(&self) -> f64 {
+        self.xmax
+    }
+
+    /// Returns the minimum y-coordinate.
+    pub fn get_ymin(&self) -> f64 {
+        self.ymin
+    }
+
+    /// Returns the maximum y-coordinate.
+    pub fn get_ymax(&self) -> f64 {
+        self.ymax
+    }
+
+    /// Returns the minimum z-coordinate, if present.
+    pub fn get_zmin(&self) -> Option<f64> {
+        self.zmin
+    }
+
+    /// Returns the maximum z-coordinate, if present.
+    pub fn get_zmax(&self) -> Option<f64> {
+        self.zmax
+    }
+
+    /// Returns the minimum m-value (measure), if present.
+    pub fn get_mmin(&self) -> Option<f64> {
+        self.mmin
+    }
+
+    /// Returns the maximum m-value (measure), if present.
+    pub fn get_mmax(&self) -> Option<f64> {
+        self.mmax
+    }
+
+    /// Returns `true` if both zmin and zmax are present.
+    pub fn is_z_valid(&self) -> bool {
+        self.zmin.is_some() && self.zmax.is_some()
+    }
+
+    /// Returns `true` if both mmin and mmax are present.
+    pub fn is_m_valid(&self) -> bool {
+        self.mmin.is_some() && self.mmax.is_some()
+    }
+}
+
+impl From<BoundingBox> for parquet::BoundingBox {
+    /// Converts our internal `BoundingBox` to the Thrift-generated format.
+    fn from(b: BoundingBox) -> parquet::BoundingBox {
+        parquet::BoundingBox {
+            xmin: b.get_xmin().into(),
+            ymin: b.get_ymin().into(),
+            xmax: b.get_xmax().into(),
+            ymax: b.get_ymax().into(),
+            zmin: b.get_zmin().map(|z| z.into()),
+            zmax: b.get_zmax().map(|z| z.into()),
+            mmin: b.get_mmin().map(|m| m.into()),
+            mmax: b.get_mmax().map(|m| m.into()),
+        }
+    }
+}
+
+impl From<parquet::BoundingBox> for BoundingBox {
+    fn from(bbox: parquet::BoundingBox) -> Self {
+        let mut new_bbox = Self::new(
+            bbox.xmin.into(),
+            bbox.xmax.into(),
+            bbox.ymin.into(),
+            bbox.ymax.into(),
+        );
+
+        new_bbox = match (bbox.zmin, bbox.zmax) {
+            (Some(zmin), Some(zmax)) => new_bbox.with_zrange(zmin.into(), zmax.into()),
+            // If both None or mismatch, set it to None and don't error
```

Comment:
Here and below, slight wording suggestion
```suggestion
            // If either None or mismatch, leave it as None and don't error
```